service-base/src/main/java/org/eclipse/hono/service/registration/BaseRegistrationService.java

```diff
@@ -212,16 +218,21 @@ public void assertRegistration(
         Objects.requireNonNull(deviceId);
         Objects.requireNonNull(resultHandler);
 
+        final Span currentSpan = newChildSpan("assert Device Registration", spanContext, tenantId, deviceId, null);
```

Comment:
this is not what I meant. What I wanted to propose was to create the span in *processAssertRequest* and pass the span instead of a `SpanContext` into this method ...

Reply:
Ok, sorry, got that wrong. (I've removed the commit again here.)
That would mean changing the `assertRegistration` method in the `RegistrationService` interface (using a Span instead of SpanContext param). And, to keep things consistent, also adapting `TenantService` (and later on `CredentialsService`).

(Thinking of having consistence in the kind of parameters used, it is to be noted here that there are similar methods in the `RegistrationClient`, `TenantClient` (and soon `CredentialsClient`) interfaces, having the `SpanContext` param there as well. As these interfaces are not necessarily implemented by users, we could keep them as is, though.)

Although I would intuitively rather refrain from passing on a span to other methods (so as to prevent the other methods from being able to finish it), I see the mentioned advantages.

I've created a new PR #962 with the corresponding changes.
(TenantService should be adapted accordingly after that.)